Add the search box and the toggle section navigation

shuuji3 opened the pull request:
✅ Resolves #20895.
📄 Page preview: https://deploy-preview-27689--kubernetes-io-master-staging.netlify.app/blog/

## Search box

This PR adds the search box to the blog page.

**Screenshot:**
<img width="859" alt="Screen Shot 2021-04-23 at 21 34 39" src="https://user-images.githubusercontent.com/1425259/115873192-9efa4080-a47d-11eb-8f19-292471c85e73.png">

## Toggle button

Also, this adds the toggle button on the right of the search box for the mobile page. Clicking the button expands the navigation menu. It's the same as the documentation pages. 

**Screenshot:**
<img width="391" alt="Screen Shot 2021-04-23 at 21 34 54" src="https://user-images.githubusercontent.com/1425259/115873262-b0434d00-a47d-11eb-8b69-ffb3742f7979.png">
<img width="387" alt="Screen Shot 2021-04-23 at 21 35 20" src="https://user-images.githubusercontent.com/1425259/115873255-ad485c80-a47d-11eb-9b60-f9fe60825238.png">

## How to test

1. Open the blog page (https://deploy-preview-27689--kubernetes-io-master-staging.netlify.app/blog).
2. Test search functionality with keywords you like.
3. Change to the mobile view.
4. Test search again.
5. Tap the toggle button and confirm to show the Table of Contents of the page.

shuuji3 commented:
/assign @kbhawkey

kbhawkey commented:
Hi @shuuji3 . The search bar looks good!
I viewed a blog page, reducing the window to about mobile size. The inner hamburger works, but when the page is displayed after the left navigation, the title for the blog page runs over the left nav (2021, 2020, 2019, ...).
I have not had time to look into the cause of the overflow.
Thanks!

shuuji3 commented:
@kbhawkey Thank you for reviewing!🙂  Although the issue didn't happen on the top page of the blog, I can also replicate the behavior when clicking the inner hamburger on each post page.

![Screen Shot 2021-04-25 at 11 41 58](https://user-images.githubusercontent.com/1425259/115978620-9ad43d00-a5bb-11eb-8263-c3ec88cc9ce7.png)

OK, I'd like to investigate the cause of this issue by my side.

shuuji3 commented:
I also found an similar issue on the left bottom of the navbar on the desktop
![Screen Shot 2021-04-25 at 12 54 12](https://user-images.githubusercontent.com/1425259/115980175-74b39a80-a5c5-11eb-9791-656b47bc4c21.png)
 view.

shuuji3 commented:
The issue should be fixed now.

**mobile view:**
![Screen Shot 2021-04-25 at 13 08 58](https://user-images.githubusercontent.com/1425259/115980934-53ee4380-a5cb-11eb-9f5f-b7f094d130aa.png)

**desktop view:**
![Screen Shot 2021-04-25 at 13 09 `06](https://user-images.githubusercontent.com/1425259/115980932-5355ad00-a5cb-11eb-932b-2c80f57f8700.png)

One style setting in the `main.sccs` is the cause of this issue. That style is needed to adjust the height of the navbar because the search bar didn't exist on the blog pages.

kbhawkey commented:
/area web-development

kbhawkey commented:
/sig blog

kbhawkey commented:
Hi @shuuji3 .
I looked at the changes again. Thanks for the updates.
I think it is worthwhile to get someone from the blog project to look over these changes.
The inner hamburger color should be the site's blue color and change colors when you hover?
Otherwise, LGTM. Thanks!
/lgtm

kbhawkey commented:
/area blog

shuuji3 commented:
@kbhawkey I investigated the issue of the inner hamburger color and fixed it. Sorry, could you take a look at it again?

The cause of the issue was the awesome font JavaScript file. It is loaded only on the blog page and replaces the hamburger `<button>` element with the `<svg>` equivalent. The style was only applied to the `<button>` hamburger, so I modified the style so that the style applies to both `<button>` and `<svg>` hamburger.

kbhawkey commented:
Looks good!
/lgtm

irvifa reviewed (commented):
/approve
